### PR TITLE
Add account unlock on password reset

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Resetting the password of an account does not lift the account lock, which is unusual behavior compared to [major online services](https://github.com/parse-community/parse-server/issues/6773#issuecomment-653702492) and therefore may cause confusion on the user side.

Related issue: closes https://github.com/parse-community/parse-server/issues/6773

### Approach
Added new account policy that allows to automatically unlock an account after successfully resetting the password.
```js
accountLockout: {
  unlockOnPasswordReset: true // Is true if the account lock should be removed after a successful password reset. Default: false
}
```
This is added as a policy option to account for different security policies, depending on use case.

### TODOs before merging

- [x] Add test cases
- [x] Add Parse Server account policy option
- [x] Add entry to changelog
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by Moumouls** [comment]
I'll take a look asap 🙂

File: src/Options/Definitions.js [hunk 1]
```diff
@@ -3,626 +3,611 @@
...
-    env: 'PARSE_SERVER_VERIFY_USER_EMAILS',
-    help: 'Enable (or disable) user email validation, defaults to false',
-    action: parsers.booleanParser,
-    default: false,
-  },
-  webhookKey: {
-    env: 'PARSE_SERVER_WEBHOOK_KEY',
-    help: 'Key sent with outgoing webhook calls',
-  },
+  "accountLockout": {
```
Comment (davimacedo) [inline src/Options/Definitions.js:9]: It is a minor item, but why do we have all this changes?

**Review by davimacedo — APPROVED** [review]
LGTM!

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js:9]: Good question, this has to be a pre-commit hook or something on my side. I think it is prettier or lint.

Comment (mtrezza) [reply src/Options/Definitions.js:9]: Oh, these changes actually seem to occur for me on `npm run definitions`. The file is then built with double quotes. Could you try this out on your side, if that happens as well?

**Comment by mtrezza** [comment]
Not merging yet, looking into [definitions quotes changes](https://github.com/parse-community/parse-server/pull/7146#discussion_r566497252).

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js:9]: It seems that the generator already adds the double quotes before writing to file, as you can see when inspecting the `res` output:

https://github.com/parse-community/parse-server/blob/25fb576776fefee631b457e358a2f567867c9c80/resources/buildConfigDefinitions.js#L279

This [babel issue](https://github.com/babel/babel/issues/4109) describes that the quotes option has also been removed and babel outputs with double quotes. So much about babel's side at least.

File: spec/AccountLockoutPolicy.spec.js [hunk 2]
```diff
@@ -340,3 +342,118 @@ describe('Account Lockout Policy: ', () => {
...
+    await expectAsync(Parse.User.logIn(username, password)).toBeRejected();
+
+    await Parse.User.requestPasswordReset(user.getEmail());
+    const link = sendPasswordResetEmail.calls.all()[0].args[0].link;
+    const linkUrl = new URL(link);
+    const token = linkUrl.searchParams.get('token');
+    const newPassword = 'newPassword';
+    await request({
+      method: 'POST',
+      url: `${config.publicServerURL}/apps/test/request_password_reset`,
```
Comment (Moumouls) [inline spec/AccountLockoutPolicy.spec.js:416]: i can suggest here to add a login try just after the `requestPasswordReset` to ensure that the lock is not removed until the reset token is used ! 🙂

Comment (mtrezza) [reply spec/AccountLockoutPolicy.spec.js]: That's a good one - added.

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js]: OK, when I run `npm run prettier` the double quotes are removed. That also affects some other files though, which are not related to this PR. What is the current workflow - should we have prettier run as a pre-commit hook? It seems that currently a PR is not guaranteed to be merged with prettier, if the author did not manually run it?

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**